internal/it/test_skip.go

```diff
@@ -0,0 +1,221 @@
+// The following conditions may be set:
+// "hz": Hazelcast version
+// "ver": go client version
+// "os": value of runtime.GOOS
+// "enterprise"/"oss": presence of enterprise key environment variable
+// Example: SkipIf(t, "ver > 1.1, hz = 5")
+func SkipIf(t *testing.T, conditionString string) {
+	skip, err := skipIf(conditionString)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	if skip {
+		t.Skip("Skipping test: SkipIf conditions met")
+	}
+}
+
+func skipIf(conditionString string) (bool, error) {
+	conditions := strings.Split(conditionString, ",")
+	for _, condition := range conditions {
+		condition = strings.Trim(condition, " ")
+		switch parts := strings.Split(condition, " "); {
+		case parts[0] == skipHzVersion:
+			if err := validateLength(parts, 3, condition, "hz = 4.0"); err != nil {
+				return true, err
+			}
+			skip, err := checkVersion(parts[2], parts[1], HzVersion())
+			if err != nil {
+				return true, err
+			}
+			if skip {
+				return true, nil
+			}
+		case parts[0] == skipClientVersion:
+			if err := validateLength(parts, 3, condition, "ver = 4.0"); err != nil {
+				return true, err
+			}
+			skip, err := checkVersion(parts[2], parts[1], internal.ClientVersion)
+			if err != nil {
+				return true, err
+			}
+			if skip {
+				return true, nil
+			}
+		case parts[0] == skipOS:
+			if err := validateLength(parts, 3, condition, "os = windows"); err != nil {
+				return true, err
+			}
+			skip, err := checkOS(parts[1], parts[2])
+			if err != nil {
+				return true, err
+			}
+			if skip {
+				return true, nil
+			}
+		case parts[0] == skipEnterprise:
+			if err := validateLength(parts, 1, condition, "enterprise"); err != nil {
+				return true, err
+			}
+			skip := checkEnterprise(false)
+			if skip {
+				return true, nil
+			}
+		case parts[0] == "!"+skipEnterprise:
+			if err := validateLength(parts, 1, condition, "!enterprise"); err != nil {
+				return true, err
+			}
+			skip := checkEnterprise(true)
+			if skip {
+				return true, nil
+			}
+		case parts[0] == skipOSS:
+			if err := validateLength(parts, 1, condition, "oss"); err != nil {
+				return true, err
+			}
+			skip := checkEnterprise(true)
+			if skip {
+				return true, nil
+			}
+		case parts[0] == "!"+skipOSS:
+			if err := validateLength(parts, 1, condition, "!oss"); err != nil {
+				return true, err
+			}
+			skip := checkEnterprise(false)
+			if skip {
+				return true, nil
+			}
+		default:
+			return true, fmt.Errorf("Unexpected test skip constant \"%s\" in %s", parts[0], condition)
+		}
+	}
+	return false, nil
+}
+
+func validateLength(parts []string, expected int, condition, example string) error {
+	if len(parts) != expected {
+		return fmt.Errorf("Unexpected format for %s, example of expected condition: \"%s\" ", condition, example)
+	}
+	return nil
+}
+
+func checkVersion(given, operator, actual string) (bool, error) {
+	isGivenGreater, err := compareVersions(given, actual)
```

Comment:
Also, would using an int instead of a pointer to bool make more sense here (*greater* variable)? Since there are three states of the comparison result.

Reply:
Addressed in 5071a2fabacf61a0c6750af893641b4dfc45f051